cypress/support/step-definitions/advanced-color-picker-steps.js

```diff
@@ -0,0 +1,61 @@
+import {
+  experimentalSimpleColorPickerInput,
+  advancedColorPickerCell,
+} from '../../locators/advanced-color-picker';
+
+const FIRST_ELEMENT = 1;
+const SECOND_ELEMENT = 2;
+
+When('I open Advanced Color Picker', () => {
+  advancedColorPickerCell().first().click();
+});
+
+When('I pick simple {int} color', (index) => {
+  if (index === 1) {
+    experimentalSimpleColorPickerInput(SECOND_ELEMENT).click();
+    experimentalSimpleColorPickerInput(FIRST_ELEMENT).click();
+  }
+  for (let i = 0; i < index; ++i) {
+    experimentalSimpleColorPickerInput(i + 1).click();
+  }
+});
+
+Then('Simple Color {int} has focus', (index) => {
+  experimentalSimpleColorPickerInput(index).should('have.focus');
+});
+
+Then('Experimental Simple Color Picker {int} element was picked up', (index) => {
+  experimentalSimpleColorPickerInput(index).should('have.attr', 'aria-checked', 'true');
+});
+
+When('I select {int} color', (index) => {
+  experimentalSimpleColorPickerInput(index).click();
+});
+
+When('I press downarrow on the {int} color', (index) => {
```

Comment:
try to use `switch` for whole keys actions. It will allow you to have one step instead of 7

Reply:
updated.